climt/_core/constants.py

```diff
@@ -0,0 +1,303 @@
+from sympl import default_constants, DataArray
+from copy import deepcopy
+
+constant_name_map = {
```

Comment:
I'd really encourage you to use the standard names within the components, instead of having a map. In the case of "solid phase" and "liquid phase" things below, you may not want to do this because they aren't necessarily for water. We have a few options for that:

- Create a new constants table in CliMT that augments the table in Sympl, and use CliMT.get_constant to access them. The downside of this is that only CliMT components would use them, which may lead to non-physical models.

- Replace the constants table in Sympl so it uses these names instead of names involving "water". The downside of this is that it is confusing to the typical Earth modeler (solid phase of what?).

- Use the constants that involve "water" in the name. I imagine that planetary modelers are used to having "water" in the model be their condensate, and just setting the properties of "water" to be the properties of their condensate. If you think a planetary modeler would see "density_of_liquid_water" and know fairly intuitively that it's the density of their condensate (and not necessarily water), assuming they set the constant value in their model script, then this would be best I think.

Reply:
This misses a few use cases, however unlikely they may be:

* Water may be a component of the atmosphere/ocean system, but may not be the major condensible (like Mars). Hence, I want to keep condensible separate from water.

* Astrophysics graduates who start their modelling career with say a Titan GCM may
not find 'water' to mean the condensible in their models.

* I don't want to replace Sympl's constants for the same reasons as your
second bullet point.

The way this constant library is written now, with the default set of constants, solid phase/ice and liquid phase/water are interchangeable. So, Sympl's constants can be used without issue for earth models.

But I think it is worth thinking about whether we want to address these issues at the Sympl level itself (since it is ultimately meant to model any planet). I am open to moving some of this code into Sympl.